libs/server/AOF/AofProcessor.cs

```diff
@@ -158,44 +167,64 @@ internal unsafe void ProcessAofRecord(IMemoryOwner<byte> entry, int length, bool
         {
             fixed (byte* ptr = entry.Memory.Span)
             {
-                ProcessAofRecordInternal(ptr, length, asReplica);
+                if (replayFromLegacyAof && IsLegacyFormat(ptr))
+                {
+                    LegacyAofHeader aofHeader = *(LegacyAofHeader*)ptr;
+                    ProcessAofRecordInternal(aofHeader.sessionID, aofHeader.opType, aofHeader.version, aofHeader.type, ptr, length, isLegacyFormat: true, asReplica);
+                }
+                else
+                {
+                    AofHeader aofHeader = *(AofHeader*)ptr;
+                    ProcessAofRecordInternal(aofHeader.sessionID, aofHeader.opType, aofHeader.version, aofHeader.type, ptr, length, isLegacyFormat: false, asReplica);
+                }
             }
             entry.Dispose();
         }
 
         /// <summary>
-        /// Process AOF record
+        /// Checks if the MSB is not set of the first pointer, this indicates it is an older AOF format
         /// </summary>
+        /// <param name="ptr"></param>
+        /// <returns></returns>
+        private static unsafe bool IsLegacyFormat(byte* ptr) => (*ptr & MsbMask) == 0;
```

Comment:
Might make more sense to have this method in AofHeader, so that logic is contained there. Then you can reuse the constants you already set there.